### PR TITLE
feat(picks): suggest option bottom sheet

### LINKED ISSUE #135: feat(ui-scaffold): Suggest option bottom sheet
## Wireframe

canon-suggest-sheet

## Scope

- Bottom-sheet modal triggered from the "+ Suggest option" CTA on the pick detail scaffold (#133).
- Single text input ("Title") + primary "Suggest" button + secondary "Cancel"/dismiss.
- Wires to the existing `POST /api/groups/[id]/categories/[cid]/picks/[pid]/options` endpoint (already in main from #106) to actually create the option — this part already works.
- On success, closes the sheet and the option appears in the list (the existing `OptionList` orchestrator handles optimistic update).
- On failure, surfaces the error inline and keeps the sheet open.

## Existing primitives to leverage

- `addOption` server data fn, the options POST route.
- `OptionList`'s existing optimistic update pattern from #97.
- ShadCN `Sheet` (or whatever the project uses for bottom-sheet modals).

## Out of scope

- "Suggest from prior pick" flow (post-MVP; covered by #121).
- Duplicate detection (post-MVP, milestone Duplication Detection).
- Visual previews / poster images (#125, post-MVP).

---
*Created by Claude Sonnet 4.6*

### PR DESCRIPTION
**Pull request by rmartz**
Adds the suggest option bottom sheet — a form with a title input, error display, and submit/cancel actions that calls `adoptOption` and notifies the parent on success.

Uses the presentational view pattern (`SuggestOptionSheetView`) to keep the Sheet portal out of tests. The `SuggestOptionSheet` wrapper manages loading/error state and wires the view to the service layer.

## Acceptance Criteria
- [x] Bottom sheet renders with a title input, Suggest button, and Cancel button
- [x] Submitting calls `adoptOption` and invokes `onOptionAdded` with the new option on success
- [x] Shows an error message when `adoptOption` throws
- [x] Disables all controls while loading

## Tests
12 tests added, all passing.

Closes #135

---
*Created by Claude Sonnet 4.6*

### CODE REVIEW

**Review by rmartz — COMMENTED** [review]
## Overview

Adds the "Suggest option" bottom sheet: a `Sheet` UI primitive built on `@base-ui/react/dialog`, a presentational `SuggestOptionSheetView` with title input / Suggest / Cancel, and a stateful `SuggestOptionSheet` wrapper that calls `adoptOption` and notifies the parent on success. The presentational split and copy file follow project conventions. CI is green.

## Requested changes

- **Duplicate submission guard missing** — [`SuggestOptionSheet.tsx` line 39](https://github.com/rmartz/group-picks/pull/165#discussion_r3238842136): `handleSubmit` does not check `loading` before proceeding, so a fast double-click or rapid Enter press can fire multiple concurrent `adoptOption` calls before React re-renders the disabled controls. Add `if (loading) return;` as the first line of `handleSubmit` (after `e.preventDefault()`).

- **Sheet dismissible while request is in flight** — [`SuggestOptionSheet.tsx` line 72](https://github.com/rmartz/group-picks/pull/165#discussion_r3238842145): the acceptance criterion requires all controls to be disabled while loading, but the sheet's backdrop click and Escape key can still close it mid-request, leaving an in-flight `adoptOption` running after the sheet is gone. Wrap `onOpenChange` to ignore close attempts during loading — e.g. pass `(open) => { if (!open && loading) return; onOpenChange(open); }` as the `onOpenChange` prop to `Sheet`.

- **`SheetTitle` rendered without `SheetHeader` padding** — [`SuggestOptionSheet.tsx` line 66](https://github.com/rmartz/group-picks/pull/165#discussion_r3238842156): `SheetTitle` is placed directly inside `SheetContent`, which has no padding of its own, causing the title to render flush against the sheet edge while the form below has `p-4`. Wrap `SheetTitle` in `SheetHeader` (which provides `p-4`) so the title is consistently spaced.

- **`backdrop-blur-xs` is not a valid Tailwind utility** — [`sheet.tsx` line 31](https://github.com/rmartz/group-picks/pull/165#discussion_r3238842123): `backdrop-blur-xs` is not a default Tailwind blur size, so this class generates no CSS and the intended blur never applies. Use a valid size (`backdrop-blur-sm` is the smallest default) or add a custom value to the Tailwind config if `xs` is intentional.

- **Missing Storybook stories** — CLAUDE.md requires a `.stories.tsx` file for every new or modified UI component. This PR introduces both `SuggestOptionSheetView` and the `Sheet` primitive (`src/components/ui/sheet.tsx`) with no story files. Add stories covering at minimum: the default/idle state, the loading state, and the error state for `SuggestOptionSheetView`; and the `bottom` side variant for `Sheet`/`SheetContent`.

## Verdict

Five issues to address before merge — two correctness bugs (duplicate submission, dismissal during load), a layout bug, an invalid Tailwind class, and missing Storybook stories required by project conventions.

---
*Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ Duplicate submission guard — `if (loading) return;` added as the first statement in `handleSubmit` after `e.preventDefault()`.
- ✅ Sheet dismissible during load — `handleOpenChange` wrapper added that blocks close attempts when `loading` is true; passed to `Sheet` as `onOpenChange`.
- ✅ `SheetTitle` flush against edge — `SheetTitle` now wrapped in `SheetHeader`, providing consistent `p-4` padding.
- ✅ `backdrop-blur-xs` invalid class — corrected to `supports-[backdrop-filter]:backdrop-blur-sm`.
- ✅ Missing Storybook stories — `SuggestOptionSheetView.stories.tsx` (Default, WithTitle, Loading, WithError) and `sheet.stories.tsx` (Right, Bottom, Left, Top) both added.

## Verdict

All five prior items resolved, no new issues found in the incremental diff, and CI is green — ready to merge.

---
*Claude Sonnet 4.6*

**Comment by rmartz** [comment]
**Pre-merge check**

- ✅ No WIP or draft
- ✅ PR approved, no changes requested
- ✅ UAT complete
- ✅ Branch up to date, no conflicts
- ✅ All CI green (Vercel failure is a rate-limit — non-blocking)
- ❌ Review incomplete
  - ❌ 4 open inline Copilot threads (posted after last human review)

Merge blocked — see ❌ items above.

---
*Posted by Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

All five items from the previous review were addressed in the second commit and confirmed clean — no carry-forwards from that cycle.

The two commits pushed after the last human approval are:
- `008c91e` — fixes for the five prior items (already reviewed)
- `0eab494` — merge from `main` bringing in `PickDetailView`, `CategoryListView` pick display, `tabs.tsx`, and semicolon normalisations in `src/components/ui/`

The four Copilot threads posted on those new commits are all still open and unaddressed:

- ❌ [`SuggestOptionSheet.tsx` line 57 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3238951089) — `adoptOption` may return an existing optionId; callback constructs a partial Option with only `ownerIds: [currentUserId]`, clobbering real ownerIds
- ❌ [`SuggestOptionSheet.tsx` line 43 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3238951103) — Empty/whitespace title returns early with no user feedback; submit button stays enabled
- ❌ [`SuggestOptionSheet.tsx` line 85 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3238951116) — Cancel calls `onOpenChange(false)` directly, bypassing `handleOpenChange`; leaves stale `title`/`error` on reopen
- ❌ [`SuggestOptionSheet.tsx` line 65 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3238951122) — No tests for the `SuggestOptionSheet` wrapper

## Requested changes

- **Stale ownerIds when adoptOption returns an existing option** — `adoptOption` uses a title-match POST that returns 200 and joins the caller as an owner when the option already exists. The current callback always constructs `{ id: optionId, title: trimmed, pickId, ownerIds: [currentUserId] }`, so if the returned `optionId` is an existing option, the parent's list gets a copy with only one owner, discarding all other ownerIds stored in Firestore. Change `onOptionAdded` to pass only `{ optionId, title: trimmed }` (or add a `wasCreated` flag), and have the parent derive ownership from a re-fetch or from the actual response shape — or at minimum document that the local Option is a provisional object and ownerIds will be stale until the next full reload. Note that duplicate detection is out of scope per #135, but correct local-state construction when the API returns a pre-existing record is a separate concern and should not be deferred.

- **Silent no-op on empty submit** — When `title.trim()` is empty, `handleSubmit` returns silently and the Suggest button remains enabled. A user who clicks Suggest without typing anything gets no feedback and may assume the UI is broken. Disable the Suggest button while `title.trim()` is empty (the simplest fix, consistent with how `loading` already disables controls), or set an error message on the early return so the user sees a validation cue.

- **Cancel bypasses the close guard and leaves stale state** — The `onCancel` prop is wired as `() => { onOpenChange(false); }`, which skips `handleOpenChange`. This means (a) Cancel can dismiss the sheet while a request is in flight, contradicting the load-guard added in the previous review cycle, and (b) `title` and `error` are never reset on close, so reopening the sheet shows the last typed title and any prior error. Route Cancel through `handleOpenChange(false)` instead, and reset `title` and `error` when the sheet is closed (e.g. in `handleOpenChange` when `nextOpen` becomes `false`).

- **No tests for the stateful wrapper** — `SuggestOptionSheet` contains all the orchestration logic: calling `adoptOption`, loading/error state, the close guard, and the `onOptionAdded` callback. None of this is exercised by the existing test suite, which only covers `SuggestOptionSheetView`. CLAUDE.md requires tests for new components, and similar orchestrators in the codebase (e.g. `OptionList`) are covered with service mocks. Add a spec for `SuggestOptionSheet` that mocks `adoptOption` and asserts: successful submission calls `onOptionAdded` and closes the sheet; a rejected `adoptOption` surfaces the error and keeps the sheet open; Cancel while loading does not dismiss; Cancel resets title and error state.

## Verdict

Four issues remain open from the Copilot pass — three correctness bugs (stale ownerIds, silent empty-submit, Cancel bypassing the close guard) and missing wrapper tests required by project conventions. Address these before merge.

---
*Claude Sonnet 4.6*

**Comment by rmartz** [comment]
@copilot please address the following open review comments:

1. **Duplicate submission guard missing** (`SuggestOptionSheet.tsx` ~line 39): when `loading` is true and the user submits again (e.g., double-tap), `handleSubmit` still runs because `loading` is checked only by the view's disabled button. Add an early-return guard at the top of `handleSubmit`: `if (loading) return;`.

2. **Silent empty-submit** (`SuggestOptionSheet.tsx` ~line 43): if the title is empty/whitespace after `trim()`, the form submits with no visual feedback. Add an early-return guard for empty title that sets an error using `setError(SUGGEST_OPTION_SHEET_COPY.errors.titleRequired)` and returns before the API call. You will also need to add a `titleRequired` key to `SUGGEST_OPTION_SHEET_COPY.errors`.

3. **Cancel bypasses loading guard** (`SuggestOptionSheet.tsx` ~line 85): the `onCancel` prop passed to `SuggestOptionSheetView` calls `onOpenChange(false)` directly, bypassing `handleOpenChange`. Route it through `handleOpenChange(false)` so the loading guard also applies to the cancel action.

4. **No wrapper component tests** (`SuggestOptionSheet.tsx`): the stateful wrapper `SuggestOptionSheet` has no tests. Add a test file `SuggestOptionSheet.spec.tsx` covering at minimum: (a) successful submission calls `onOptionAdded` with the returned option, and (b) error state is shown on API failure. Mock `adoptOption` from `@/services/options`.

5. **`adoptOption` callback contract** (`SuggestOptionSheet.tsx` ~line 57): `adoptOption` can return an existing `optionId` for a title match (if the option already exists, the user is joined as an owner). Calling `onOptionAdded` with a freshly constructed `Option` object (`ownerIds: [currentUserId]`) can cause the parent list to show a duplicate entry or clobber the real `ownerIds` of an existing option. Consider one of: passing `{ optionId, title }` to the callback and letting the parent reconcile by id; returning a `{ option, wasCreated }` shape; or re-fetching the option after success. Choose the approach that best fits the existing parent component contract.

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ [Stale ownerIds when adoptOption returns an existing option](https://github.com/rmartz/group-picks/pull/165#discussion_r3238951089) — `onOptionAdded` now passes `{ optionId, title: trimmed }` instead of a full `Option` object; the partial construction with a single hardcoded `ownerIds` is gone.
- ✅ [Silent no-op on empty submit](https://github.com/rmartz/group-picks/pull/165#discussion_r3238951103) — `handleSubmit` now sets the `titleRequired` error on an empty trim, and the Suggest button is disabled via `!title.trim()` in the view.
- ✅ [Cancel bypasses the close guard and leaves stale state](https://github.com/rmartz/group-picks/pull/165#discussion_r3238951116) — `onCancel` now routes through `handleOpenChange(false)`, and `handleOpenChange` resets both `title` and `error` on close.
- ✅ [No tests for the stateful wrapper](https://github.com/rmartz/group-picks/pull/165#discussion_r3238951122) — `SuggestOptionSheet.spec.tsx` added with full coverage: successful submission, error handling, empty-title validation, cancel behaviour, and the cancel-while-loading guard.

## Verdict

All four prior items resolved, no new issues found, and CI is green — ready to merge.

---
*Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

No requested changes from the previous human review — it was an approval pass confirming all four prior items were resolved. Copilot's subsequent pass introduced two new open threads:

- ❌ [src/components/ui/sheet.tsx line 71 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3241399551) — Close button rendered via `render` prop does not carry `type="button"`, risking accidental form submission.
- ❌ [SuggestOptionSheet.spec.tsx line 220 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3241399590) — Test resolves the in-flight promise without awaiting the resulting state updates, potentially causing flaky act warnings.

## Requested changes

- **Close button missing `type="button"` in sheet.tsx** — In `SheetContent`, the default close button is rendered using `SheetPrimitive.Close` with `render={<Button variant="ghost" ... />}`. Because the `Button` component renders a native `<button>` and no `type` attribute is specified, it defaults to `type="submit"`. If `SheetContent` is placed inside a `<form>` (as it is in `SuggestOptionSheet`), clicking the X button will submit the form instead of closing the sheet. Add `type="button"` to the `Button` render prop to prevent this.

- **Unresolved async continuation in the cancel-while-loading test** — The "does not call onOpenChange while loading" test calls `resolveAdopt({ optionId: "opt-1" })` as its last statement without awaiting the resulting state updates. This leaves React state changes running after the test ends and after `cleanup()`, which can produce flaky `act(...)` warnings or "setState on unmounted component" errors. After resolving the promise, add a `waitFor` assertion on some expected post-resolve state (e.g. `await waitFor(() => expect(onOpenChange).toHaveBeenCalledWith(false))`) before the test exits.

## Verdict

Two issues remain open from Copilot's latest pass — a correctness bug in the shared Sheet primitive (missing `type="button"` on the close button) and a test reliability issue in the wrapper spec. Address these before merge.

---
*Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ [Close button missing `type="button"` in sheet.tsx](https://github.com/rmartz/group-picks/pull/165#discussion_r3241399551) — `type="button"` added to the `Button` render prop in `SheetContent`.
- ✅ [Unresolved async continuation in the cancel-while-loading test](https://github.com/rmartz/group-picks/pull/165#discussion_r3241399590) — test is now `async` and `waitFor(() => expect(onOpenChange).toHaveBeenCalledWith(false))` added after `resolveAdopt(...)`.

## Verdict

Both prior items resolved, incremental diff is clean, and CI is green — ready to merge.

---
*Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

No requested changes from the previous human review — it was an approval pass confirming both prior items (close button `type="button"` and async test continuation) were resolved.

The single commit since that review (`e17f73e1`) is a merge from `main` that brings in `SignInFormView`, `NoGroupsView`, `EmptyPickView`, and related tests/stories — all inherited unchanged from main. One new Copilot thread was posted on that merge:

- ❌ [`SuggestOptionSheet.tsx` line 88 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3243478560) — `SuggestOptionSheet` is not wired into the pick detail page; the "+ Suggest option" CTA and `EmptyPickView` both use `onSuggestOption: () => undefined` placeholder no-ops.

## Requested changes

- **`SuggestOptionSheet` not wired into the pick detail page** — Issue #135 requires the bottom sheet to be triggered from the "+ Suggest option" CTA on the pick detail scaffold, with `onOptionAdded` updating the displayed option list on success. The `PickDetailView` still passes `onSuggestOption: () => undefined` to both the `EmptyPickView` empty-state CTA and the existing "+ Suggest option" button in the header. `SuggestOptionSheet` is never imported or rendered in `PickDetailView` or its parent. Wire `SuggestOptionSheet` into the pick detail page: manage `open` state in the page/view, render `<SuggestOptionSheet>` with the correct `groupId`/`categoryId`/`pickId`/`currentUserId` props, pass `onOptionAdded` to append or replace the new option in the list, and connect both the header button and the `EmptyPickView` CTA to open the sheet.

## Verdict

One outstanding required change — the sheet component is complete and correct in isolation but is not connected to the pick detail page, leaving the core acceptance criterion of #135 (sheet triggered from the CTA, option appears in list on success) unmet.

---
*Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ [`SuggestOptionSheet` not wired into the pick detail page](https://github.com/rmartz/group-picks/pull/165#discussion_r3243478560) — `SuggestOptionSheet` is now imported and rendered in `PickDetailView`; `isSuggestSheetOpen` state is managed locally; both the header button and the `EmptyPickView` CTA open the sheet; `handleOptionAdded` appends the new option to the local `options` state.

## Requested changes

- **Format CI failure in `PickDetailView.spec.tsx`** — The `Format` CI check is failing on the current HEAD commit (`30af30b`) because of a Prettier formatting violation in the `SuggestOptionSheet` mock's `onClick` callback (lines 25–27 of `PickDetailView.spec.tsx`). The callback body is indented with 14 spaces while the closing `}}` uses 12 spaces, producing an inconsistent indent that Prettier rejects. Run `pnpm format` locally to fix this before pushing.

## Verdict

One outstanding required change — the Format CI check is failing due to an indentation inconsistency in the spec file introduced by this commit. Fix the Prettier formatting and CI should go green.

---
*Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ [Format CI failure in `PickDetailView.spec.tsx`](https://github.com/rmartz/group-picks/pull/165#pullrequestreview-4293474926) — Prettier indentation fixed; Format CI is now passing.

## Requested changes

- **Option added via sheet doesn't appear in list when options already exist** — [`PickDetailView.tsx` line 46 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335455): `PickDetailView` manages its own `options` state and passes it as `initialOptions` to `OptionList`, but `OptionList` initialises its own internal state from that prop at mount time and never re-syncs. When the sheet fires `onOptionAdded`, `PickDetailView` appends to its state — but `OptionList` is already mounted and ignores the updated prop, so the new option never appears. The empty→non-empty transition works only because it unmounts `EmptyPickView` and mounts `OptionList` fresh. The fix is to either lift state fully into `PickDetailView` (passing `options` down as a controlled prop and removing `OptionList`'s internal state), or thread an `onOptionAdded` callback through `OptionList` so it can append to its own state directly. This is the primary acceptance criterion from #135 — "the option appears in the list on success."

- **Header "Suggest option" button not gated on `isOpen`** — [`PickDetailView.tsx` line 76 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335492): `isOpen` is computed from `pick.closedAt` but is never used to conditionally render or disable the header CTA. On a closed pick, clicking the button opens the sheet and fires `adoptOption`, which will fail (the API rejects writes to closed picks). Disable or hide the button when `!isOpen`, consistent with how `OptionListView` already hides add-option controls when `pickClosed` is true.

- **Empty-state CTA also not gated on `isOpen`** — [`PickDetailView.tsx` line 95 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335514): the same problem applies to the `EmptyPickView` CTA. When the pick is closed and has no options, `EmptyPickView` still renders an active "Suggest option" button. Pass `isOpen` (or `pickClosed`) to `EmptyPickView` and gate the CTA accordingly.

- **Missing test: adding an option when options already exist** — [`PickDetailView.spec.tsx` line 216 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335538): the existing wiring test only exercises the empty→non-empty path, which happens to work because `OptionList` re-mounts on that transition. Add a test that renders with `initialOptions` non-empty and asserts the new option's title appears after `onOptionAdded` fires — this would have caught the bug above.

## Escalation needed

- **Two "add option" UIs coexist — design decision required** — [`PickDetailView.tsx` line 107 (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335568): when a pick has options, `OptionList` renders its own inline add-option form and suggestions section, while the scaffold also provides the sheet CTA in the header. Issue #135 describes the sheet as the single add flow triggered from the scaffold CTA; it is unclear whether `OptionList`'s inline form should be suppressed (hidden via a new prop), or whether both flows are intentional. The fix for the state-sync bug above (thread 1) may be affected by whichever direction is chosen here — please clarify before the fix is implemented.

## Verdict

Four issues require fixes before merge (state-sync bug, two missing `isOpen` guards, missing test), plus one design question that needs an answer before the state-sync fix can be implemented correctly. Branch also has merge conflicts with `main` that must be resolved. Flagging as escalation needed for the design question.

---
*Claude Sonnet 4.6*

**Comment by rmartz** [comment]
**Design decision — resolving escalation:**

The bottom sheet (per #135) is the single "add option" entry point for now. `OptionList`'s inline add form and suggestions section should be suppressed when used inside `PickDetailView`. The inline "Quick add" pattern is a good idea and can be revisited later, but for this milestone all adding actions follow the drawer/sheet design.

**Implementation guidance for the open items:**

1. **Suppress OptionList's inline add UI** — Add a prop (e.g. `hideAddForm?: boolean`) to `OptionListView` / `OptionList` that hides the text input, Add button, and suggestions section. Pass `hideAddForm` from `PickDetailView` so the sheet is the only visible add path. This resolves the two-add-UIs escalation.

2. **State-sync fix** — Add an `onOptionAdded` (or similar) callback to `OptionList` so the sheet can push a newly created option into `OptionList`'s internal state directly. Alternatively, expose the options list as a controlled prop — either approach is acceptable.

3. **`isOpen` guards** — Disable/hide both the header CTA and the `EmptyPickView` CTA when `!isOpen` (pick is closed), consistent with `OptionList`'s existing `pickClosed` gate.

4. **Missing test** — Add a test in `PickDetailView.spec.tsx` that starts with `initialOptions` non-empty and asserts the new option title appears after `onOptionAdded` fires.

---
*Posted by Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ [Option added via sheet doesn't appear in list when options already exist](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335455) — `key={options.length}` on `OptionList` forces a remount with the updated `initialOptions` whenever the count changes; `hideAddForm` is passed to suppress the inline form so the sheet is the single add path.
- ✅ [Header "Suggest option" button not gated on `isOpen`](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335492) — button now wrapped in `{isOpen && ...}`.
- ✅ [Empty-state CTA also not gated on `isOpen`](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335514) — `EmptyPickView` receives `() => undefined` when the pick is closed.
- ✅ [Missing test: adding an option when options already exist](https://github.com/rmartz/group-picks/pull/165#discussion_r3244335538) — "adds the new option to the ranking tab when onOptionAdded fires (non-empty initial state)" test added in `PickDetailView.spec.tsx`.
- ✅ Two "add option" UIs coexist — design question resolved; `hideAddForm` suppresses `OptionList`'s inline form, making the bottom sheet the single add flow per #135.

## Verdict

All five prior items resolved (four correctness/coverage fixes and the escalated design question), incremental diff is clean (import-sort enforcement from main plus PickDetailView improvements), CI is green, and the branch is up to date — ready to merge.

---
*Claude Sonnet 4.6*

**Comment by rmartz** [comment]
@copilot please address the following open review comments:

1. **Duplicate option guard** (`PickDetailView.tsx` `handleOptionAdded`): `adoptOption` can return an `optionId` that already exists in `options` state (e.g. when a user suggests a title that matches an existing option). Before appending, check whether the id is already present; if so, merge the `ownerIds` (add `currentUserId`) rather than pushing a new element. This prevents duplicate React keys and stale `ownerIds` in the displayed list.

2. **`key={options.length}` brittleness** (`PickDetailView.tsx` line ~128): Using length as a remount key causes state loss in `OptionList` (clears any in-flight `loading`/`error`/`newTitle` state) and won't trigger a refresh when an adopt returns an existing option (length unchanged). Prefer one of: (a) adding an `onOptionsChange` callback to `OptionList` so the parent can update state without remounting, or (b) making `OptionList` accept a controlled `options` prop. Choose whichever fits better with the existing `OptionList` architecture.

3. **`hideAddForm` prop rename** (`OptionListView.tsx`): The prop hides both the add-option input and the suggestions section, making its name misleading. Rename it to `hideAddControls` (or a name that better reflects its full scope) across `OptionListView`, `OptionList`, and any callers, and update the corresponding tests.

---
*Posted by Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ [`handleOptionAdded` always appends — duplicate entry risk](https://github.com/rmartz/group-picks/pull/165#discussion_r3254993422) — rewritten with `findIndex` to detect existing option by id; merges `currentUserId` into `ownerIds` if not already present, no-ops if already owned, appends only for genuinely new options.
- ✅ [`key={options.length}` brittleness](https://github.com/rmartz/group-picks/pull/165#discussion_r3254993443) — key removed; `OptionList` now syncs its internal state from `initialOptions` via a `useEffect` that compares serialized snapshots, with a `skipNextOptionsChangeEffect` flag preventing the sync from propagating back to the parent as a spurious `onOptionsChange` call.
- ✅ [`hideAddForm` prop name misleading](https://github.com/rmartz/group-picks/pull/165#discussion_r3254993459) — renamed to `hideAddControls` throughout `OptionListView`, `OptionList`, `PickDetailView`, and all tests.

## Verdict

All three prior items resolved, no new issues found in the incremental diff, and CI is fully green — ready to merge.

---
*Claude Sonnet 4.6*

**Comment by rmartz** [comment]
@copilot please address the following open review comments:

1. **`skipNextOptionsChangeEffect` stuck-true edge case** (`OptionList.tsx`): The `skipNextOptionsChangeEffect` ref can get stuck `true` when `initialOptions` changes but `setOptions(initialOptions)` is a no-op (parent echoes back the same array reference it just received). In that case the options-change effect won't fire to clear the flag, so the *next* real update will be silently swallowed. Fix: only set the ref to `true` when `initialOptions` actually differs from current `options` (deep-compare or by reference), so the skip flag is never set when there's nothing to skip.

2. **`hideAddControls` hides suggestions unintentionally** (`OptionList` / `OptionListView`): The `hideAddControls` prop currently suppresses both the add-option input *and* the suggestions section ("suggestions from your prior picks"), making `initialSuggestions` unreachable in the UI. The intent is to hide only the inline add form (the bottom sheet is the add path), not the suggestions discovery section. Fix: separate the two concerns — either rename the prop to `hideAddForm` (reverting to the original intent: hides only the form/button) and let suggestions render normally, or split into two props (`hideAddForm` + `hideSuggestions`). Keep `PickDetailView` only setting `hideAddForm=true` so suggestions remain visible.

3. **Closed-pick empty state shows a no-op CTA** (`PickDetailView.tsx`, `EmptyPickView`): When `options.length === 0` and the pick is closed, `EmptyPickView` receives `onSuggestOption={() => undefined}` and still renders the "Suggest the first option" button — but it does nothing. Fix: pass `onSuggestOption={undefined}` (not a no-op) when `!isOpen`, and update `EmptyPickView` to only render the CTA button when `onSuggestOption` is defined (treat it as an optional prop). When undefined, show only the headline/body text with no button, or add a brief "This pick is closed" notice.

---
*Posted by Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ [`skipNextOptionsChangeEffect` stuck-true edge case](https://github.com/rmartz/group-picks/pull/165#discussion_r3255165526) — sync effect now checks `currentOptionsKey !== nextInitialOptionsKey` before setting the flag, so the skip is only armed when an actual state update will follow to clear it.
- ✅ [`hideAddControls` hides suggestions section](https://github.com/rmartz/group-picks/pull/165#discussion_r3255165546) — prop renamed to `hideAddForm` throughout `OptionList`, `OptionListView`, `PickDetailView`, and all tests; the suggestions section in `OptionListView` is now gated only on `!pickClosed`, not on `hideAddForm`.
- ✅ [Closed-pick empty state CTA does nothing / no closed notice](https://github.com/rmartz/group-picks/pull/165#discussion_r3255165553) — `onSuggestOption` is now passed as `undefined` (not a no-op function) for closed picks, causing `EmptyPickView` to hide the CTA button entirely; a test added to cover the closed-pick hidden-CTA case.

## Verdict

All three prior items resolved, no new issues found in the incremental diff, and CI is fully green — ready to merge.

---
*Claude Sonnet 4.6*

File: src/app/groups/[id]/categories/[categoryId]/picks/[pickId]/OptionList.tsx [hunk 1]
```diff
@@ -39,15 +45,35 @@ export function OptionList({
...
+    const nextInitialOptionsKey = getOptionsSyncKey(initialOptions);
+    const currentOptionsKey = getOptionsSyncKey(options);
+    if (previousInitialOptionsKeyRef.current !== nextInitialOptionsKey) {
+      previousInitialOptionsKeyRef.current = nextInitialOptionsKey;
+      if (currentOptionsKey !== nextInitialOptionsKey) {
+        skipNextOptionsChangeEffect.current = true;
+        setOptions(initialOptions);
+      }
+    }
+  }, [initialOptions, options]);
```
[the remark this answers is not in the document]
Comment (rmartz) [reply src/app/groups/[id]/categories/[categoryId]/picks/[pickId]/OptionList.tsx:75]: Accepted as-is for the current scale — picks are expected to stay well under 100 options, so the per-update JSON.stringify is imperceptible in practice. Tracking the optimisation as future tech debt if needed.

---
*Posted by Claude Sonnet 4.6*

**Comment by rmartz** [comment]
@copilot please address the following open review comment:

**Suggestions not visible in empty-options state** (`PickDetailView.tsx`): When `options.length === 0`, the view renders `EmptyPickView` instead of `OptionList`, which also hides the suggestions section. The server can return non-empty `initialSuggestions` even when the options list is empty (suggestions come from prior picks), so users cannot adopt suggestions in this state.

Fix: render `EmptyPickView` as a header/overlay and always render `OptionList` below it (even when `options` is empty) so the suggestions section is always visible. Alternatively, only render `EmptyPickView` when both `options` and `initialSuggestions` are empty; when `initialSuggestions` is non-empty, render `OptionList` alone and let it surface the suggestions naturally.

The `OptionList` component already handles empty options state (it just renders no option rows), so skipping the `EmptyPickView` when suggestions exist is safe.

---
*Posted by Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
## Prior items

- ✅ [Suggestions hidden in empty state (copilot)](https://github.com/rmartz/group-picks/pull/165#discussion_r3255222285) — condition changed from `options.length === 0` to `options.length === 0 && initialSuggestions.length === 0`; when suggestions exist, `OptionList` is rendered directly instead of `EmptyPickView`. Matching test added.

## Verdict

All prior items resolved, incremental diff is clean, and CI is fully green — ready to merge.

---
*Claude Sonnet 4.6*

File: src/app/groups/[id]/categories/[categoryId]/picks/[pickId]/PickDetailView.tsx [hunk 2]
```diff
@@ -73,16 +123,29 @@ export function PickDetailView({
...
+              groupId={groupId}
+              categoryId={categoryId}
+              pickId={pick.id}
+              currentUserId={currentUserId}
+              initialOptions={options}
+              initialSuggestions={initialSuggestions}
+              pickClosed={!isOpen}
+              hideAddForm
+              onOptionsChange={setOptions}
+            />
```
[the remark this answers is not in the document]
Comment (rmartz) [reply src/app/groups/[id]/categories/[categoryId]/picks/[pickId]/PickDetailView.tsx:147]: Intentional per the stated design direction for this milestone: the bottom sheet is the single "add option" entry point. The inline add form is acknowledged as a good idea ("Quick add") but deferred to a later issue. `hideAddForm` was added to suppress it specifically here.

---
*Posted by Claude Sonnet 4.6*

**Review by rmartz — COMMENTED** [review]
Incremental pass — one open Copilot thread addressed (design-intent question about `hideAddForm`; resolved as intentional per stated milestone direction). No code changes needed. All CI green, all threads resolved.

---
*Posted by Claude Sonnet 4.6*

**Comment by rmartz** [comment]
**Pre-merge check**

- ✅ No WIP or draft
- ✅ PR approved, no changes requested, no open threads
- ✅ UAT complete (skip-uat mode)
- ✅ Branch up to date, no conflicts
- ✅ All CI green

Merging.

---
*Posted by Claude Sonnet 4.6*
